src/commands/buffer.rs

```diff
@@ -126,7 +126,11 @@ impl Buffer {
             bail!("Invalid size for buffer: {}", size);
         }
 
+        let len = self.data_buffer.len();
         self.data_buffer.resize(size, 0);
+        if len > 8192 && size < len {
+            self.data_buffer.shrink_to_fit();
```

Comment:
I'm not sure if it's worth the effort, but might be better to use the [`Vec#shrink_to` function](https://doc.rust-lang.org/1.45.2/std/vec/struct.Vec.html#method.shrink_to) _if_ it is available. It's currently only available on nightly and it's not clear when/if it will get stabilized.

Reply:
Yeah, shrink_to would indeed be better, but I think this is fine for now and I wanted to make sure that it would be fixed for anyone who runs on stable as well.

Reply:
Might be worth putting a //TODO in there, dunno.

Reply:
Agree that the priority should be to fix the issue on stable. Not really sure how you would use `shrink_to` or `shrink_to_fit` conditionally based on whether the feature is available or not? Think you're probably right and we should just add a TODO to optimize this later if/when the feature becomes available on stable rust.